File: go/coupon_event_data.go
// this file is @generated
package meteroid

import "time"

type CouponEventData struct {
	Code string `json:"code"`

	CouponId CouponId `json:"coupon_id"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description string `json:"description"`

	Disabled bool `json:"disabled"`

	Discount CouponDiscount `json:"discount"`

	// RFC 3339 timestamp.
	ExpiresAt *time.Time `json:"expires_at,omitempty"`

	RecurringValue *int32 `json:"recurring_value,omitempty"`

	RedemptionLimit *int32 `json:"redemption_limit,omitempty"`

	Reusable bool `json:"reusable"`
}

File: python/meteroid/models/coupon_event_data.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .coupon_discount import CouponDiscount
from .coupon_id import CouponId


@dataclasses.dataclass
class CouponEventData(BaseModel):
    code: str

    coupon_id: CouponId

    created_at: datetime

    description: str

    disabled: bool

    discount: CouponDiscount

    reusable: bool

    expires_at: t.Optional[datetime] = None

    recurring_value: t.Optional[int] = None

    redemption_limit: t.Optional[int] = None
